packages/block-editor/src/components/block-list/block.js

```diff
@@ -53,6 +61,51 @@ import useHoveredArea from './hover-area';
 import { isInsideRootBlock } from '../../utils/dom';
 import useMovingAnimation from './moving-animation';
 
+const EMPTY_OBJECT = {};
+function useMetaAttributeSourceBackwardsCompatibility(
+	name,
+	_attributes,
+	_setAttributes
+) {
+	const { attributes: attributeTypes = EMPTY_OBJECT } =
+		getBlockType( name ) || EMPTY_OBJECT;
+	let [ attributes, setAttributes ] = [ _attributes, _setAttributes ];
+
+	if ( Object.values( attributeTypes ).some( ( type ) => type.source === 'meta' ) ) {
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		const [ meta, setMeta ] = useEntityProp( 'postType', 'post', 'meta' );
+
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		attributes = useMemo(
+			() => ( {
+				..._attributes,
+				...Object.keys( attributeTypes ).reduce( ( acc, key ) => {
+					if ( attributeTypes[ key ].source === 'meta' ) {
+						acc[ key ] = meta[ attributeTypes[ key ].meta ];
+					}
+					return acc;
+				}, {} ),
+			} ),
+			[ attributeTypes, meta, _attributes ]
+		);
+
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		setAttributes = useCallback(
+			( ...args ) => {
+				Object.keys( args[ 0 ] ).forEach( ( key ) => {
+					if ( attributeTypes[ key ].source === 'meta' ) {
+						setMeta( { [ attributeTypes[ key ].meta ]: args[ 0 ][ key ] } );
+					}
+				} );
+				return _setAttributes( ...args );
+			},
+			[ attributeTypes, setMeta, _setAttributes ]
+		);
+	}
+
+	return [ attributes, setAttributes ];
+}
```

Comment:
Can we somehow bail early if we detect that the blockType doesn't have any "meta" attribute (for performance reasons).

Reply:
Yeah, it does that:

```js
if ( Object.values( attributeTypes ).some( ( type ) => type.source === 'meta' ) ) {
```